### PR TITLE
User Flow

### PR DESCRIPTION
**Pull request by VladyslavTodorchuk**

### CODE REVIEW

File: app/concepts/v1/session/operation/create.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  step :find_user
+  step :validates_password
+  step :generate_tokens
+
+  def find_user(ctx, params:, **)
+    ctx[:user] = User.find_by(username: params[:username])
+  end
+
+  def validates_password(_ctx, params:, user:, **)
+    user&.valid_password?(params[:password])
```
Comment (CaptainJNS) [inline app/concepts/v1/session/operation/create.rb:11]:
```suggestion
    user.valid_password?(params[:password])
```

File: app/concepts/v1/session/operation/create.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class V1::Session::Operation::Create < Trailblazer::Operation
+  step :find_user
+  step :validates_password
+  step :generate_tokens
```
Comment (CaptainJNS) [inline app/concepts/v1/session/operation/create.rb:4]: what about fails?

File: app/concepts/v1/user/operation/create.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  def validates_password?(ctx, user:, params:, **)
+    ctx[:param] = :password
+    user.valid_password?(params[:password_confirmation])
+  end
+
+  def create_user(ctx, params:, **)
+    ctx[:user] = User.new(username: params[:username], password: params[:password])
+  end
```
Comment (CaptainJNS) [inline app/concepts/v1/user/operation/create.rb:20]: define methods in order of steps calling

File: app/controllers/api/v1/sessions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  def destroy
+    result = V1::Session::Operation::Destroy.call(payload: payload)
+
+    if result.success?
+      render status: :ok
+    else
+      render status: :not_found
+    end
```
Comment (CaptainJNS) [inline app/controllers/api/v1/sessions_controller.rb:21]: use guard clause

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  PASSWORD_FORMAT = /[a-zA-Z]*/.freeze
+
+  validates :username, allow_blank: false, length: { minimum: MIN_USERNAME_LENGTH, maximum: MAX_USERNAME_LENGTH }
+  validates :password, allow_blank: false, length: { minimum: MIN_PASSWORD_LENGTH },
+                       format: { with: PASSWORD_FORMAT,
+                                 message: 'Password length should be 8 characters, alphanumeric.' }
+
+  def valid_password?(confirmation_password)
+    password.match(confirmation_password)
+  end
```
Comment (CaptainJNS) [inline app/models/user.rb:15]: You don't need to verify the match of passwords in API. It's a front-end responsibility. You even don't need the confirmation password parameter

File: config/routes.rb [hunk 6]
```diff
@@ -1,3 +1,12 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  namespace :api do
+    namespace :v1 do
+      resources :users, only: %i[create destroy]
+      resources :refresh, only: :create
+
+      delete 'sing_out/destroy_by_refresh', to: 'sessions#destroy_by_refresh'
+      post 'sing_in', to: 'sessions#create'
+      delete 'sing_out', to: 'sessions#destroy'
```
Comment (CaptainJNS) [inline config/routes.rb:9]:
```suggestion
      delete 'sign_out/destroy_by_refresh', to: 'sessions#destroy_by_refresh'
      post 'sign_in', to: 'sessions#create'
      delete 'sign_out', to: 'sessions#destroy'
```

Comment (CaptainJNS) [inline config/routes.rb:9]: Use REST:
`resource :session, only: %i[create destroy]`

File: spec/controllers/api/v1/sessions_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        post :create, params: { username: user.username, password: user.password }
+      end
+
+      it 'has status success' do
+        expect(response).to have_http_status(:created)
+      end
+
+      it 'has return json with tokens' do
+        expect(JSON.parse(response.body).keys.sort).to eq %w[access access_expires_at csrf refresh
+                                                             refresh_expires_at]
```
Comment (CaptainJNS) [inline spec/controllers/api/v1/sessions_controller_spec.rb:18]:
```suggestion
        expect(JSON.parse(response.body).keys).to match_array %w[access access_expires_at csrf refresh
                                                             refresh_expires_at]
```

**Review by CaptainJNS — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 5 above)
Comment (VladyslavTodorchuk) [reply app/models/user.rb:15]: i user it to check user password that came from front to back for authentication.

**Pull request merged**